### PR TITLE
feat/AB#62227 Implement Select Component

### PR DESCRIPTION
**Pull request by Bricecrx**
# Description

The feature added to the library is a select menu linked to the new ui library designed. The select menu has four inputs : multiselect to know if multiselection is allowed, selectTriggerTemplate that gives a template or a simple string to use as the trigger of the select menu, disabled to know if the select menu should be disabled and finally the list of options that can be selected. It also has 3 outputs : opened (to know if the listbox is opened), closed and selectedOption (list of the selected option(s)).

## Ticket

I can't paste the azure ticket link because I do not have access to Azure. Here is the ticket id however : [AB#62227](https://dev.azure.com/WHOHQ/2888b816-2d79-49ef-a86e-561e6cd456a0/_workitems/edit/62227)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

I used storybook to test the visual of the component created and the impact of the parameters used.

## Screenshots

image
image
image
image
image
image
image

# Checklist:

( * == Mandatory ) 

- [x] * My code follows the style guidelines of this project
- [x] * Linting does not generate new warnings
- [x] * I have performed a self-review of my own code
- [x] * I have commented my code, particularly in hard-to-understand areas
- [x] * I have put JSDoc comment in all required places
- [x] * My changes generate no new warnings
- [x] * I have included screenshots describing my changes if relevant
- [x] * I have selected labels in the Pull Request, according to the changes with code brings

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -8,5 +8,8 @@
   "i18n-ally.keystyle": "nested",
   "i18n-ally.usage.scanningIgnore": ["widgets/*", "dist/*"],
   "i18n-ally.tabStyle": "tab",
-  "i18n-ally.indent": 4
+  "i18n-ally.indent": 4,
+  "cSpell.words": [
```
Comment (AntoineRelief) [inline .vscode/settings.json]: @Bricecrx what is this change ?

File: libs/ui/src/lib/select-menu/select-menu.component.stories.ts [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+  selectMenu: new FormControl(),
+});
+
+/**
+ * Template for standalone selection select menu
+ *
+ * @param args args
+ * @returns story of select menu component
+ */
+const TemplateStandaloneSelection: Story<SelectMenuComponent> = (
```
Comment (AntoineRelief) [inline libs/ui/src/lib/select-menu/select-menu.component.stories.ts]: @unai-reliefapp @Bricecrx the 'story' is deprecated
we need to use storyobj

**Comment by AntoineRelief** [comment]
@Bricecrx @unai-reliefapp 
a few things about this one:
- some changes are not clear
- the options should not be an input. I do like the Angular approach that uses a custom component: mat-option, to do the job
- the label is not useful in the select, I removed it. This is material

File: .vscode/settings.json (hunk 1 above)
Comment (Bricecrx) [reply .vscode/settings.json]: No idea, I think it is automatic, maybe when importing cdkListbox because I can't remember to have modify vscode/settings

**Pull request merged**